inter/block.go

```diff
@@ -1,45 +1,188 @@
 package inter
 
 import (
-	"github.com/Fantom-foundation/lachesis-base/hash"
+	"math/big"
+	"unsafe"
+
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/trie"
 )
 
+// Block represents the on-disk storage format of a block. It contains all
+// fields required to reconstruct the block header, as well as a list of
+// hashes of the transactions been executed as part of the represented block.
+//
+// This struct should be considered immutable. No fields should be modified,
+// directly or indirectly. Ideally, all fields should be private, but that
+// would invalidate support for RLP encoding as it is used to store instances
+// on the disk. However, future updates may make fields inaccessible.
+//
+// To create a new block, use the BlockBuilder, handling the computation of
+// key properties implicitly.
 type Block struct {
-	Time        Timestamp
-	Atropos     hash.Event
-	Events      hash.Events
-	Txs         []common.Hash // non event txs (received via genesis or LLR)
-	InternalTxs []common.Hash // DEPRECATED in favor of using only Txs fields and method internal.IsInternal
-	SkippedTxs  []uint32      // indexes of skipped txs, starting from first tx of first event, ending with last tx of last event
-	GasUsed     uint64
-	Root        hash.Hash
+	// Fields required for the block header.
+	Number               uint64
+	ParentHash           common.Hash
+	StateRoot            common.Hash
+	Time                 Timestamp
+	Difficulty           uint64
+	GasLimit             uint64
+	GasUsed              uint64
+	BaseFee              *big.Int
+	PrevRandao           common.Hash
+	TransactionsHashRoot common.Hash
```

Comment:
what is the difference with transactionHashes? 
is this hash redundant when considering the parentHash and the transactionHashes?

Reply:
The `TransactionHashes` are a list of of hashes of the transactions being part of a given block. They are used to reference transactions being stored in a different table in case the full block needs to be restored.

The `TransactionsHashRoot` is an aggregated hash over all transactions that requires the list of all (full) transactions to be recomputed. It could be recomputed based on the list of `TransactionHashes` if there is access to the transaction table.

When requesting the header of a block, only the `TransactionHashRoot` is required, as it commits all contained transactions and their order. I think it is a sensible improvement to store the root explicitly for faster access.